### PR TITLE
Review

### PR DESCRIPTION
**Pull request by samoht**
/cc @talex5

### CODE REVIEW

File: src/ivfs/ivfs_tree.ml [hunk 1]
```diff
@@ -235,28 +234,16 @@ module Make (Store: STORE) = struct
...
-                lookup_dir t d >|= fun t ->
-                (t, path @ [d]) :: acc
-              | `File      , f ->
-                fn (path @ [f]) (fun () -> lookup_file t f) >|= fun () ->
-                acc
+          map t >>= fun childs ->
+          let childs = String.Map.bindings childs in
+          Lwt_list.fold_left_s (fun acc (k, v) ->
+              match v with
+              |  `Directory  t -> Lwt.return ((t, path @ [k]) :: acc)
```
Comment (talex5) [inline src/ivfs/ivfs_tree.ml]: (two extra spaces in this line)

File: src/ivfs/ivfs_tree.ml [hunk 2]
```diff
@@ -267,8 +254,7 @@ module Make (Store: STORE) = struct
     let diff x y =
       let set t =
         let acc = ref KVSet.empty in
-        iter t (fun k v ->
-            v () >>= fun (v, _) ->
+        iter t (fun k (v, _) ->
```
Comment (talex5) [inline src/ivfs/ivfs_tree.ml]: Do we care about changes to file types here too (normal/link/exec)?
(this PR doesn't change anything; just a comment)

**Comment by talex5** [comment]
LGTM

**Comment by samoht** [comment]
File permissions are now taken into account in the Tree.diff function.

**Pull request merged**